src/module/iotjs_module_buffer.c

```diff
@@ -119,6 +119,33 @@ static size_t bound_range(int index, size_t low, size_t upper) {
 }
 
 
+static unsigned hex2bin(char c) {
+  if (c >= '0' && c <= '9')
+    return c - '0';
+  if (c >= 'A' && c <= 'F')
+    return 10 + (c - 'A');
+  if (c >= 'a' && c <= 'f')
+    return 10 + (c - 'a');
+  return (unsigned)(-1);
+}
+
+
+static size_t hex_decode(char* buf, size_t len, const char* src,
+                         const size_t srcLen) {
+  size_t i;
+
+  for (i = 0; i < len && i * 2 + 1 < srcLen; ++i) {
+    unsigned a = hex2bin(src[i * 2 + 0]);
+    unsigned b = hex2bin(src[i * 2 + 1]);
+    if (!~a || !~b)
```

Comment:
Not sure that is a safe way to check it. Might be worth to change the return value of `hex2bin` to signed int.